packages/dev-server-esbuild/src/EsbuildPlugin.ts

```diff
@@ -164,17 +166,23 @@ export class EsbuildPlugin implements Plugin {
     target: string | string[],
   ): Promise<string> {
     try {
-      const { code: transformedCode, warnings } = await transform(code, {
+
+      const tsconfigRaw = this.esbuildConfig.tsconfig ? await promisify(fs.readFile)(this.esbuildConfig.tsconfig, 'utf8') : undefined;
```

Comment:
This will read the config each transform. Let's:
- Do promisify(fs.readFile) only once on top
- Read the ts config in server start